testslide/mock_callable.py

```diff
@@ -366,8 +366,12 @@ async def run(self, *args, **kwargs):
 
 
 class _ReturnValueRunner(_Runner):
-    def __init__(self, target, method, original_callable, value):
+    def __init__(self, target, method, original_callable, value, allow_coro=False):
         super().__init__(target, method, original_callable)
+        if not allow_coro and _is_coroutine(value):
+            raise ValueError(
+                "Setting coroutines as return value is not recommended. Use mock_async_callable instead"
+            )
```

Comment:
allowing coroutine returns from sync contexts is not nice i think, mock_async_callable would work for those cases. allow_coro is only introduced, because mock_callable and mock_async callable share the same codebase

Reply:
I'd say so in the message then, leaving no room for interpretation, like "Setting coroutines as return value is not allowed. Use mock_async_callable instead"